Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -87,29 +88,56 @@ class TabManagerImplementation: LegacyTabManager {
     // MARK: - Save tabs
 
     override func preserveTabs() {
-        guard AppConstants.useNewTabDataStore else {
-            super.preserveTabs()
+        /// For now we want to continue writing to both data stores so that we can revert to the old system if needed
+        super.preserveTabs()
+        guard shouldUseNewTabStore() else {
             return
         }
 
-        // TODO: FXIOS-6123 Handle new save logic
+        Task {
+            /// This value should never be nil but we need to still treat it as if it can be nil until the old code is removed
```

Comment:
Same thing for orphaned doc comment
```suggestion
            // This value should never be nil but we need to still treat it as if it can be nil until the old code is removed
```